### PR TITLE
feat: recovery transaction processing state

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves [missing "Processing..." state for recovery transactions](https://www.notion.so/safe-global/Do-we-have-a-processing-relaying-state-instead-of-awaiting-confirmation-for-the-Account-recovery-5a07f9ed7cde429ba976fe7ad16e5f59)

## How this PR fixes it

A new recovery-focused event bus has been added that matches that of the one for "standard" transactions and messages. Using that, a new pending map is updated according to the dispatched events and parsed for the current status of a recovery attempt.

## How to test it

1. Queue a recovery attempt.
2. Execute/cancel the attempt and observe the new processing status.

Note: data loading should also be tested - recovery state updates when a proposal is executed, cancelled and/or settings are changed.

## Screenshots

image

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/recovery/ExecuteRecoveryButton/index.tsx [hunk 1]
```diff
@@ -49,13 +52,18 @@ export function ExecuteRecoveryButton({
 
       trackError(Errors._812, e)
       setSubmitError(err)
+    } finally {
+      setPending((prev) => {
+        const { [recovery.transactionHash]: _, ...rest } = prev
+        return rest
+      })
```
[the remark this answers is not in the document]
Comment (iamacook) [reply src/components/recovery/ExecuteRecoveryButton/index.tsx:59]: I've taken the opportunity to refactor the transactional logic of recovery into an event bus that matches that of our "standard" transactions and messages. This helped fix this issue, as well as cleanup a lot of logic.

File: src/components/recovery/RecoveryStatus/index.tsx [hunk 2]
```diff
@@ -1,12 +1,34 @@
...
+  [PendingStatus.SUBMITTING]: 'Submitting',
+  [PendingStatus.PROCESSING]: 'Processing',
+}
 
 export const RecoveryStatus = ({ recovery }: { recovery: RecoveryQueueItem }): ReactElement => {
   const { isExecutable, isExpired } = useRecoveryTxState(recovery)
+  const { pending } = useContext(RecoveryContext)
+
+  const status = pending?.[recovery.transactionHash] ? (
+    STATUS_LABELS[pending[recovery.transactionHash]]
```
Comment (usame-algan) [inline src/components/recovery/RecoveryStatus/index.tsx:21]: As discussed I think we need to use `recovery.args.txHash` here instead.

File: src/components/recovery/RecoveryStatus/index.tsx [hunk 3]
```diff
@@ -1,12 +1,34 @@
...
 import type { ReactElement } from 'react'
 
 import ClockIcon from '@/public/images/common/clock.svg'
 import { useRecoveryTxState } from '@/hooks/useRecoveryTxState'
 import type { RecoveryQueueItem } from '@/services/recovery/recovery-state'
+import { PendingStatus } from '@/store/pendingTxsSlice'
+import { RecoveryContext } from '../RecoveryContext'
+
+const STATUS_LABELS: Partial<Record<PendingStatus, string>> = {
+  [PendingStatus.SUBMITTING]: 'Submitting',
```
Comment (usame-algan) [inline src/components/recovery/RecoveryStatus/index.tsx:12]: Should we use the `RecoveryEvent` instead? It doesn't have the Submitting event though (not sure if there is an equivalent)

File: src/components/recovery/RecoveryContext/useRecoveryState.ts [hunk 4]
```diff
@@ -35,34 +33,10 @@ export function useRecoveryState(delayModifiers?: Array<Delay>): {
...
-      safeAddress,
-      safe.chainId,
-      safe.version,
-    ],
-    false,
-  )
+  // Reload recovery data when a Guardian transaction occurs
+  useEffect(() => {
+    return recoverySubscribe(RecoveryEvent.PROCESSED, refetch)
+  }, [refetch])
```
Comment (usame-algan) [inline src/components/recovery/RecoveryContext/useRecoveryState.ts:39]: With this in place, do we still need the tx history listener?

File: src/hooks/useRecoveryTxState.ts [hunk 5]
```diff
@@ -24,8 +30,9 @@ export function useRecoveryTxState({ validFrom, expiresAt, transactionHash, args
   const isNext =
     !delayModifier || (sameAddress(delayModifier.address, address) && args.queueNonce.eq(delayModifier.txNonce))
   const isExecutable = isNext && isValid && !isExpired
+  const isPending = !!pending?.[transactionHash]
```
Comment (usame-algan) [inline src/hooks/useRecoveryTxState.ts:33]: I think we also have to change it here to `args.txHash`

File: src/components/recovery/RecoveryStatus/index.tsx (hunk 2 above)
Comment (iamacook) [reply src/components/recovery/RecoveryStatus/index.tsx]: The has was changed in c8a056f7ef0229bcfd4dc12036622ca21ef5ebbf and the spinner was added in b2fca640c6d6d7aea467fda126a3e0ffb97b26af.

File: src/components/recovery/RecoveryContext/useRecoveryState.ts (hunk 4 above)
Comment (iamacook) [reply src/components/recovery/RecoveryContext/useRecoveryState.ts:39]: Yes, the history listener is required for cancellations as we do not know what kind of transaction is executed.

File: src/components/recovery/RecoveryStatus/index.tsx (hunk 3 above)
Comment (iamacook) [reply src/components/recovery/RecoveryStatus/index.tsx:12]: As discussed: this will be left as is as it is not event but pending status related.

**Review by usame-algan — APPROVED** [review]

**Pull request merged**